### PR TITLE
Refactor display

### LINKED ISSUE #73: Implement KV table

### LINKED ISSUE #82: Make Tooltips work for KV

### PR DESCRIPTION
**Pull request by hannessolo**
Closes #82 
Closes #73 

This pull request refactores the graph display to eliminate code duplication and makes hover over nodes and the tables work for all types types of database (need to implement interface, already done for eventlog and keyvalue).

### CODE REVIEW

File: src/adapters/OrbitDBNodeProvider.ts [hunk 1]
```diff
@@ -118,7 +118,7 @@ export class OrbitDBNodeProvider implements NodeProvider {
      
     let ipfsLog = operationsLog.getInnerLog();
     index.updateIndex(ipfsLog);
-    let result = index.get();
-    return result;
+    console.log(index);
```
Comment (justincqz) [inline src/adapters/OrbitDBNodeProvider.ts]: Unsure if proper to leave in debug messages.

File: src/components/OrbitDBStoreDisplay.tsx [hunk 2]
```diff
@@ -118,12 +108,7 @@ const OrbitDBEventStoreDisplay: React.FC<{
    * @param operationsLog The operations log used to reconstruct the database state.
    */
   function reconstructData(operationsLog: OperationsLog): Array<any> {
-
-    let index = new EventIndex();
-    let ipfsLog = operationsLog.getInnerLog();
-    index.updateIndex(ipfsLog);
-    let result = index.get();
-    return result;
+    return nodeProvider.reconstructData(operationsLog);
```
Comment (justincqz) [inline src/components/OrbitDBStoreDisplay.tsx]: Should we simply not have this method (if all it does is wrap the Node Provider method?)

File: src/components/viewDatabase/Sidebar.tsx [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+const SideBar: React.FC<{
+  joinEvents: string[],
+  selectJoin(join: string): void,
+  type: string,
+  store: Store,
+  goHome(): void,
+  uiProvider: DatabaseUIProvider
+}> = ({ joinEvents, selectJoin, type, goHome, uiProvider, store }) => {
+  const [showAddFields, toggleAddFields] = useState(false);
```
Comment (justincqz) [inline src/components/viewDatabase/Sidebar.tsx]: Might have to make this more generic as we gonna have other buttons soon

Comment (justincqz) [reply src/components/viewDatabase/Sidebar.tsx]: I can handle it so you can simply resolve it if you don't feel like changing

Comment (hannessolo) [reply src/components/viewDatabase/Sidebar.tsx]: So you mean the goHome bit?

File: src/components/OrbitDBStoreDisplay.tsx (hunk 2 above)
Comment (hannessolo) [reply src/components/OrbitDBStoreDisplay.tsx]: Good point

File: src/adapters/OrbitDBNodeProvider.ts (hunk 1 above)
Comment (hannessolo) [reply src/adapters/OrbitDBNodeProvider.ts]: Yah lemme fix

File: src/components/viewDatabase/Sidebar.tsx (hunk 3 above)
Comment (justincqz) [reply src/components/viewDatabase/Sidebar.tsx]: Nah the showAddFields bit

**Pull request merged**